File: laravel/app/Models/Item.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Item extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<string>
     */
        protected $fillable = [
            'user_id',
            'category_id',
            'title',
            'description',
            'condition',
            'estimated_value',
            'currency',
            'location_city',
            'location_lat',
            'location_lon',
            'wants_description',
            'status',
        ];
    
        /**
         * Get the user that owns the item.
         */
        public function user(): BelongsTo
        {
            return $this->belongsTo(User::class);
        }
    
        /**
         * Get the category of the item.
         */
        public function category()
        {
            return $this->belongsTo(Category::class);
        }
    
        /**
         * Get the images for the item.
         */
        public function images()
        {
            return $this->hasMany(ItemImage::class);
        }
    
        /**
         * Get the desired categories for the item.
         */
        public function wants()
        {
            return $this->hasMany(ItemWant::class);
        }
    
        /**
         * Get all swipes on this item.
         */
        public function swipesOnThisItem()
        {
            return $this->hasMany(Swipe::class, 'swiped_on_item_id');
        }
    
        /**
         * Get all swipes made with this item.
         */
        public function swipesWithThisItem()
        {
            return $this->hasMany(Swipe::class, 'swiper_item_id');
        }
    
        /**
         * Get all matches where this item is item A.
         */
        public function matchesAsItemA()
        {
            return $this->hasMany(BarterMatch::class, 'item_a_id');
        }
    
        /**
         * Get all matches where this item is item B.
         */
        public function matchesAsItemB()
        {
            return $this->hasMany(BarterMatch::class, 'item_b_id');
        }
    
        /**
         * Get all matches for this item.
         */
        public function matches()
        {
            return $this->matchesAsItemA()->union($this->matchesAsItemB()->toBase());
        }
    }
    

File: Flutter/lib/models/item.dart
import 'package:trade_match/models/category.dart';
import 'package:trade_match/models/item_image.dart';
import 'package:trade_match/models/item_want.dart';

class Item {
  final int id;
  final int userId;
  final int categoryId;
  final String title;
  final String description;
  final String condition;
  final double? estimatedValue;
  final String currency;
  final String locationCity;
  final double locationLat;
  final double locationLon;
  final String? wantsDescription;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final Category? category;
  final List<ItemImage>? images;
  final List<ItemWant>? wants;

  Item({
    required this.id,
    required this.userId,
    required this.categoryId,
    required this.title,
    required this.description,
    required this.condition,
    this.estimatedValue,
    required this.currency,
    required this.locationCity,
    required this.locationLat,
    required this.locationLon,
    this.wantsDescription,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.category,
    this.images,
    this.wants,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: int.parse(json['id'].toString()),
      userId: int.parse(json['user_id'].toString()),
      categoryId: int.parse(json['category_id'].toString()),
      title: json['title'],
      description: json['description'],
      condition: json['condition'],
      estimatedValue: json['estimated_value'] != null ? double.parse(json['estimated_value'].toString()) : null,
      currency: json['currency'],
      locationCity: json['location_city'],
      locationLat: double.parse(json['location_lat'].toString()),
      locationLon: double.parse(json['location_lon'].toString()),
      wantsDescription: json['wants_description'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      category: json['category'] != null
          ? Category.fromJson(json['category'])
          : null,
      images: json['images'] != null
          ? (json['images'] as List)
              .map((image) => ItemImage.fromJson(image))
              .toList()
          : null,
      wants: json['wants'] != null
          ? (json['wants'] as List)
              .map((want) => ItemWant.fromJson(want))
              .toList()
          : null,
    );
  }
}
